client/landing/stepper/declarative-flow/tailored-ecommerce-flow.ts

```diff
@@ -46,13 +48,13 @@ function getPlanFromRecurType( recurType: string ) {
 const ecommerceFlow: Flow = {
 	name: ECOMMERCE_FLOW,
 	isSignupFlow: true,
-	useSignupStartEventProps() {
+	useTracksEventProps() {
 		const recur = useSelect(
 			( select ) => ( select( ONBOARD_STORE ) as OnboardSelect ).getEcommerceFlowRecurType(),
 			[]
 		);
 
-		return { recur };
+		return useMemo( () => ( { [ STEPPER_TRACKS_EVENT_SIGNUP_START ]: { recur } } ), [ recur ] );
```

Comment:
I think we should save people who don't use `useMemo` by serializing the object somehow before passing it to the effect.

Reply:
I have the same feeling. By definition though, `useTracksEventProps` being a hook, it can have internal state or rerun/react to changes in external state (such a store), so it will force rerenders up the tree as per the natural life cycle. 

I'm really not too fond of "hacking" away from the normal data flow in React. Perhaps we need to accept the reality and ask consumers to be smart about it. Otherwise, we can turn the hook to a regular function and that should make it clear that it won't be deviant to rerunning on state changes. 

I think that's more or less my thought process in general

Reply:
Oh I meant the above as in more generally about the risk involved (as in connection to your comment in the other PR)

We can certainly do some sort of string-serialization on framework end to avoid same object not being memoized. Just the reality doesn't change that state changing from here will propagate up